concordium-sdk/src/main/java/com/concordium/sdk/CurrencyConverter.java

```diff
@@ -0,0 +1,137 @@
+package com.concordium.sdk;
+
+import com.concordium.sdk.requests.smartcontracts.Energy;
+import com.concordium.sdk.responses.chainparameters.ChainParameters;
+import com.concordium.sdk.transactions.CCDAmount;
+import com.concordium.sdk.types.ConversionResult;
+import com.concordium.sdk.types.UInt64;
+
+import static com.concordium.sdk.types.ConversionResult.*;
+
+/**
+ * Utility class for converting between {@link CCDAmount}, {@link Energy} and Euros.
+ * Represents converted values using {@link ConversionResult} to ensure precision.
+ */
+public class CurrencyConverter {
+
+    /**
+     * Converts {@link Energy} to euro using exchange rate from the provided {@link ChainParameters}.
+     *
+     * @param energy {@link Energy} to convert.
+     * @param parameters {@link ChainParameters} with exchange rate used for conversion.
+     * @return {@link ConversionResult} corresponding to the euro value of {@link Energy}.
+     */
+    public static ConversionResult<EUR> energyToEuro(Energy energy, ChainParameters parameters) {
+        ConversionResult<NRG> energyFraction = from(energy.getValue(), UInt64.from(1));
+        return energyToEuro(energyFraction, parameters);
+    }
+
+    /**
+     * Converts {@link ConversionResult} representing {@link Energy} to euro using exchange rate from the provided {@link ChainParameters}.
+     *
+     * @param energy {@link ConversionResult} representing an amount of {@link Energy} to convert.
+     * @param parameters {@link ChainParameters} with exchange rate used for conversion.
+     * @return {@link ConversionResult} corresponding to the euro value of {@link Energy}.
+     */
+    public static ConversionResult<EUR> energyToEuro(ConversionResult<NRG> energy, ChainParameters parameters) {
+        ConversionResult<ConversionRate> euroPerEnergy = from(parameters.getEuroPerEnergy());
+        return energy.mult(euroPerEnergy);
+    }
+
+    /**
+     * Converts {@link Energy} to micro CCD using exchange rate from the provided {@link ChainParameters}.
+     *
+     * @param energy {@link Energy} to convert.
+     * @param parameters {@link ChainParameters} with exchange rate used for conversion.
+     * @return {@link ConversionResult} corresponding to the micro CCD value of {@link Energy}.
+     */
+    public static ConversionResult<CCD> energyToMicroCCD(Energy energy, ChainParameters parameters) {
```

Comment:
Why is this called "toMicroCCD" when the response is meant to be a CCD amount?

Reply:
It is a fraction representing the amount of microCCD returned. I've renamed the phantom type so the returntype is `ConversionResult<microCCD>`

Reply:
The `CCDAmount` class used in the SDK represents CCD as microCCD. I figured it was for the best to make it explicit that the returned amount was in micro CCD